internal/service/event/main.go

```diff
@@ -0,0 +1,102 @@
+package event
+
+import (
+	"fmt"
+
+	"github.com/rarimo/geo-points-svc/internal/config"
+	"github.com/rarimo/geo-points-svc/internal/data"
+	"github.com/rarimo/geo-points-svc/internal/data/evtypes"
+	"github.com/rarimo/geo-points-svc/internal/data/evtypes/models"
+	"github.com/rarimo/geo-points-svc/internal/data/pg"
+	"github.com/rarimo/geo-points-svc/internal/service/handlers"
+)
+
+func Run(cfg config.Config, date int) error {
+	log := cfg.Log()
+	db := cfg.DB()
+	lvls := cfg.Levels()
+	evTypes := cfg.EventTypes()
+
+	balancesQ := pg.NewBalances(db)
+	eventsQ := pg.NewEvents(db)
+	referralsQ := pg.NewReferrals(db)
+
+	evType := evTypes.Get(models.TypeEarlyTest, evtypes.FilterInactive, evtypes.FilterByAutoClaim(true))
+	if evType == nil {
+		log.Infof("Event type %s is inactive", models.TypeEarlyTest)
+		return nil
+	}
+
+	balances, err := balancesQ.FilterByCreatedBefore(date).FilterVerified().Select()
+
+	if err != nil {
+		log.WithError(err).Error("failed to filter by updated before")
+		return err
+	}
+	if len(balances) == 0 {
+		log.Infof("no balances found")
+		return nil
+	}
+
+	nullifiers := make([]string, 0)
+
+	for _, balance := range balances {
+		nullifiers = append(nullifiers, balance.Nullifier)
+	}
+
+	filteredEvents, err := eventsQ.
+		FilterByType(models.TypeEarlyTest).
+		FilterByStatus(data.EventFulfilled).
+		FilterByNullifier(nullifiers...).
+		Select()
```

Comment:
You shouldn't use filter by type, because user can have already claimed or opened events so you add second event for user.

Reply:
@kish1n remove filter by status